benchmarks/dlibqueue.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <utility>
+#include <dlib/pipe.h>
+#include "wrappers.h"
+
+template<typename T>
+struct DlibQueueWrapper
+{
+public:
+    typedef DummyToken producer_token_t;
+    typedef DummyToken consumer_token_t;
+
+public:
+    DlibQueueWrapper() : q(1024*1024*1024) { }
```

Comment:
Hmm, preallocating space for a billion elements seems a little unfair. At the same time, the documentation says that if `enqueue` is called and the queue is full, it will block until room is available, which would cause some benchmarks to deadlock.

Reply:
this can actually hurt the performance of the data structure as well, because all 99% of the time enqueues are hitting uncached memory.

Reply:
```dlib::pipe<>``` is a bounded blocking queue. So the max capacity needs to be high to make it a fair comparison to the other unbounded non-blocking queues. You might say that the benchmarks aren't really designed for blocking queues.

Reply:
Agreed!